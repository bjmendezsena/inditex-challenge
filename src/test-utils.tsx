import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import type { RenderResult } from "@testing-library/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { theme } from "./theme";

export const testQueryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      refetchOnMount: true,
      cacheTime: 0,
    },
  },
  logger: {
    log: console.log,
    warn: console.warn,
    // âœ… no more errors on the console for tests
    error: process.env.NODE_ENV === "test" ? () => {} : console.error,
  },
});

export const queryWrapper = ({ children }: { children: React.ReactNode }) => (
  <QueryClientProvider client={testQueryClient}>{children}</QueryClientProvider>
);

const DefaultTheme = extendTheme(theme);

export const render = (ui: JSX.Element, theme = DefaultTheme): RenderResult => {
  const wrapper = ({ children }: { children?: React.ReactNode }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return testingLibraryRender(ui, { wrapper });
};
