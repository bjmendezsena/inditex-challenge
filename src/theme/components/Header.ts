import type { HeaderThemeConfiguration } from "../../components";

export const ConfigHeader: HeaderThemeConfiguration = {
  parts: ["container", "contentContainer"],
  baseStyle: {
    container: {
      zIndex: 10,
      bg: "white",
      height: "64px",
      pl: 6,
      pr: 8,
      py: 4,
      boxShadow: "0px 3px 9px -7px rgba(0,0,0,0.75)",
    },
    contentContainer: {
      maxW: "1400px",
      justifyContent: "space-between",
      margin: "auto",
      w: "full",
    },
  },
};
