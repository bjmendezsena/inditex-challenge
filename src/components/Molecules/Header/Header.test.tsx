import { screen } from "@testing-library/react";
import { render } from "../../../test-utils";
import type { HeaderProps } from "./Header";
import { Header } from "./Header";
import { Stack } from "@chakra-ui/react";

describe(`<Header />`, () => {
  const defaultProps: HeaderProps = {
    leftContent: (
      <Stack data-testid='left-content' spacing={6}>
        <h1>Left</h1>
      </Stack>
    ),
    rightContent: (
      <div data-testid='right-content'>
        <h1>Right</h1>
      </div>
    ),
  };
  const factoryComponent = (props: HeaderProps = defaultProps) =>
    render(<Header {...props} />);

  test("renders correctly", () => {
    const view = factoryComponent();
    expect(screen.getByTestId("header")).toBeInTheDocument();
    expect(view.asFragment()).toMatchSnapshot();
  });
  test("renders correctly only left content", () => {
    const view = factoryComponent({
      leftContent: defaultProps.leftContent,
    });
    expect(screen.getByTestId("left-content")).toBeVisible();
    expect(screen.queryByTestId("right-content")).toBeNull();
    expect(view.asFragment()).toMatchSnapshot("OnlyLeftContent");
  });

  test("renders correctly only right content", () => {
    const view = factoryComponent({
      rightContent: defaultProps.rightContent,
    });
    expect(screen.getByTestId("right-content")).toBeVisible();
    expect(screen.queryByTestId("left-content")).toBeNull();
    expect(view.asFragment()).toMatchSnapshot("OnlyRightContent");
  });
});
