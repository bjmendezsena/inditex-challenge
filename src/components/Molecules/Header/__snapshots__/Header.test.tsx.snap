// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1gzl4dy"
    data-testid="header"
  >
    <div
      class="css-1k3povc"
    >
      <div
        class="chakra-stack css-ahthbn"
        data-testid="left-content"
      >
        <h1>
          Left
        </h1>
      </div>
      <div
        data-testid="right-content"
      >
        <h1>
          Right
        </h1>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Header /> renders correctly only left content: OnlyLeftContent 1`] = `
<DocumentFragment>
  <div
    class="css-1gzl4dy"
    data-testid="header"
  >
    <div
      class="css-1k3povc"
    >
      <div
        class="chakra-stack css-ahthbn"
        data-testid="left-content"
      >
        <h1>
          Left
        </h1>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Header /> renders correctly only right content: OnlyRightContent 1`] = `
<DocumentFragment>
  <div
    class="css-1gzl4dy"
    data-testid="header"
  >
    <div
      class="css-1k3povc"
    >
      <div
        class="css-17xejub"
      />
      <div
        data-testid="right-content"
      >
        <h1>
          Right
        </h1>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
